### PR TITLE
Support for dynamic port allocation

### LINKED ISSUE #128: Support for dynamic port allocation
To be able to start my postgres in a test environment where other tests for other services may run in parallel I'd like to dynamically allocate the port and retrieve the actually allocated port when postgres is started. net.Listen() already supports dynamic port allocation. This is done by passing the port number 0. When I tried using port 0 with embedded-postgres I got this error message:

> waiting for server to start....2024-01-26 08:39:40.670 GMT [47041] FATAL:  0 is outside the valid range for parameter "port" (1 .. 65535)
 stopped waiting
pg_ctl: could not start server

### PR DESCRIPTION
**Pull request by mainbrain**
Support for dynamic port allocation

resolves #128

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -137,10 +136,6 @@ func (c Config) BinaryRepositoryURL(binaryRepositoryURL string) Config {
 	return c
 }
 
-func (c Config) GetConnectionURL() string {
```
Comment (fergusstrange) [inline config.go:140]: We won't be able to remove this without bumping a major version, which I think we're unlikely to want to do for this change. Can you try and work it into here?

**Review by fergusstrange — CHANGES_REQUESTED** [review]
Hey @mainbrain these changes wouldn't be backwards compatible meaning we'd need to do a major version bump, which I'd like to avoid. Can you have a go working the changes to be backwards compatible? Cheers

**Comment by fergusstrange** [comment]
Hey @mainbrain I'll take a look again this weekend if you've made above requested changes.

File: config.go (hunk 1 above)
Comment (mainbrain) [reply config.go:140]: I already pitched a solution attempt within config.go and we both agreed that there should be no major logic there.

**Comment by mainbrain** [comment]
> Hey @mainbrain these changes wouldn't be backwards compatible meaning we'd need to do a major version bump, which I'd like to avoid. Can you have a go working the changes to be backwards compatible? Cheers

I pushed another variant. Maybe that's more to your liking. The trade off is: We are backwards compatible but the interface is less consistent, so I had to write more documentation.

**Pull request closed without merging**